packages/gatsby/src/utils/changed-pages.ts

```diff
@@ -0,0 +1,54 @@
+import { boundActionCreators } from "../redux/actions"
+const { deletePage, deleteComponentsDependencies } = boundActionCreators
+
+import { isEqualWith, IsEqualCustomizer } from "lodash"
+import { IGatsbyPage } from "../redux/types"
+
+export function deleteUntouchedPages(
+  currentPages: Map<string, IGatsbyPage>,
+  timestamp: number
+): string[] {
+  const deletedPages: string[] = []
+
+  // Delete pages that weren't updated when running createPages.
+  currentPages.forEach(page => {
+    if (
+      !page.isCreatedByStatefulCreatePages &&
+      page.updatedAt < timestamp &&
```

Comment:
Can you add a comment noting that `timestamp` is the result of `Date.now()`?

Reply:
It's subtly different though. It's not _now_ now, it's the time before we ran the APIs. I'll give it a better name.

Reply:
For me it was more about that being returned by _a_ `Date.now()` call